packages/ui/src/components/Menu/SMenuItem.vue

```diff
@@ -39,10 +35,10 @@ const submenuApi = useSubmenuApi()
 if (submenuApi) {
   isInSubmenu.value = true
 
-  submenuApi.register(props.index)
+  submenuApi.register(ref(props.index))
 
   onUnmounted(() => {
-    submenuApi.unregister(props.index)
+    submenuApi.unregister(ref(props.index))
```

Comment:
It is not gonna work... Created ref is not updated when `props.index` is updated. You should pass `computed(() => props.index)`.